probtorch/stochastic.py

```diff
@@ -321,6 +324,30 @@ def log_joint(self, sample_dim=None, batch_dim=None, nodes=None):
                 log_prob = log_prob + log_p
         return log_prob
 
+    def log_pair(self, sample_dim, batch_dim, nodes):
+        """Return the log average probability values, optionally for a subset of nodes.
+
+        Arguments:
+            nodes: The subset of nodes for which the log pair probabilities to be computed.
+            sample_dim(int): The dimension that enumerates samples.
+            batch_dim(int): The dimension that enumerates batch items.
+        """
+
+        log_avg_joint = 0.0
+        log_avg_sep = 0.0
+        # Note: not entirely sure if this is correct for more than one node
```

Comment:
Lol. This is the code review equivalent of writing "If you read this, then I will give you a bottle of Champagne" in your thesis.